emission/analysis/modelling/trip_model/util.py

```diff
@@ -1,7 +1,49 @@
 from typing import List, Tuple
 from past.utils import old_div
+from math import radians, cos, sin, asin, sqrt
 import numpy
 from numpy.linalg import norm
+import pandas as pd
+
+import emission.core.wrapper.confirmedtrip as ecwc
+
+
+def get_replacement_mode_features(feature_list, dependent_var, is_prediction, trips: ecwc.Confirmedtrip) -> List[float]:
+    """extract the features needed to perform replacement mode modeling from a set of
+    trips.
+
+    recodes variables that are categorical, and (TODO: scales numeric variables 0-1).
+
+    :param feature_list: features to gather from each trip
+    :type feature_list: List[string]
+    :param dependent_var: the feature to predict for each trip
+    :type dependent_var: string
+    :param is_prediction: whether or not to extract the dependent var
+    :type is_prediction: bool
+    :param trips: all trips to extract features from
+    :type trips: List[ecwc.Confirmedtrip]
+    :return: the training X features and y for the replacement mode model
+    :rtype: Tuple[List[List[float]], List[]]
+    """
+    # get dataframe from json trips; fill in calculated columns
+    trips_df = pd.json_normalize(trips)
+    # distance
+    trips_coords = trips_df[['data.start_loc.coordinates','data.end_loc.coordinates']]
+    trips_df['distance_miles'] = trips_coords.apply(lambda row : haversine(row[0],row[1]), axis=1)
```

Comment:
each trip already has its distance in meters. you should just use it directly instead of calculating the haversine, which is more intensive, and which does not even take the trajectory into count.

Reply:
Also, this may be a pet peeve, but why are we using `miles` internally? Sure, the American public is stuck in the imperial system, but when we do science under the hood, surely we can use SI units.

Reply:
Changed, you have my full and earnest agreement.